utils/ioutils.go

```diff
@@ -0,0 +1,59 @@
+package utils
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+var ErrShortWrite = errors.New("short write")
+
+type asyncMultiWriter struct {
+	writers []io.Writer
+}
+
+// AsyncMultiWriter creates a writer that duplicates its writes to all the
+// provided writers asynchronous
+func AsyncMultiWriter(writers ...io.Writer) io.Writer {
+	w := make([]io.Writer, len(writers))
+	copy(w, writers)
+	return &asyncMultiWriter{w}
+}
+
+// write data asynchronous by each writer and wait for go routine to complete
+// in case of an error write will not complete
+func (t *asyncMultiWriter) Write(p []byte) (int, error) {
+	var wg sync.WaitGroup
+	wg.Add(len(t.writers))
+	errChannel := make(chan error)
+	finished := make(chan bool, 1)
+	for _, w := range t.writers {
+		/* write data in async */
+		go writeData(p, w, &wg, errChannel)
+	}
+	/* wait group in the go routine we ensure either all pass */
```

Comment:
Let's move to go-style comments - // instead of /*
In addition, this comment is incorrect in terms of grammar.

Reply:
This comment was removed.